File: mario-less/credit/credit.c
#include <cs50.h>
#include <stdio.h>

int string_length(char s[])
{
    int c = 0;
    while (s[c] != '\0')
        c++;

    return c;
}

int charToInt(char c)
{
    int num = 0;

    // Substract '0' from entered char to get
    // corresponding digit
    num = c - '0';

    return num;
}

int main(void)
{
    string cardNumber = get_string("Number: ");
    // int count;
    // for (int i = 0; i++;)
    // {
    //     count = 0;

    //     if (cardNumber[i] != "\0")
    //     {
    //         count++;
    //     }
    //     else
    //     {
    //         break;
    //     }
    // }

    int lengthString = string_length(cardNumber);

    int total = 0;
    for (int i = lengthString - 2; i >= 0; i -= 2)
    {
        total += charToInt(cardNumber[i]) * 2;
        
        }

    printf("%i", total);
}

File: week6/sentimental-credit/credit.py
from cs50 import get_int, get_string

def main():
    cardNumber = 0
    while cardNumber <0:
        try:
            cardNumber = get_int("Number: ")
        except:
            cardNumber =-1
            continue
    lengthCard = len(cardNumber)
    firstNumber = int(cardNumber[0])
    secondNumber = int(cardNumber[1])
    total = 0
    for(i in range(lengthCard-2,-1,-2)):
        temp = 2 * int(cardNumber[i])
        if temp > 10:
            temp -=10
            total +=1
        total += temp
    for(i in range(lengthCard-1,-1,-2)):
        total += int(cardNumber[i])

    if total !=20:
        print('INVALID\N')
        return

    



main()